### ISSUE TITLE
Check the accuracy of collide_from_below()

### ISSUE THREAD

**Issue by campreilly** [campreilly/UnderSeaModelingLibrary#146]
This method has a comment it in that says, "need to rectify this code, it mimics the bottom_reflection but signs should be changed to adjust for approaching from below."

We may be able to simplify both collide_from_below() and collide_from_above() if we assume that the volume scattering layer is nearly flat so that a modified version of surface reflection can be used.

**Comment by Tibonium** [comment]
This is in reference to the computation of the grazing angle. While the grazing angle is always calculated for an interaction with the bottom, the same is not done for interactions with the surface.

To allow volume collisions to produce full eigenverbs, the grazing angle is needed when colliding from below a volume layer. I simply copied the logic for computing grazing angle in the bottom reflection case into this function and then added a note to verify that I did not have a sign error in the future. (Volumes have not yet been set to be tested and so the importance of this note was lower than current issues)
